torch/distributions/constraints.py

```diff
@@ -456,16 +460,44 @@ def check(self, value):
         return _LowerCholesky().check(value) & unit_row_norm
 
 
-class _PositiveDefinite(Constraint):
+class _Square(Constraint):
     """
-    Constrain to positive-definite matrices.
+    Constrain to square matrices.
     """
     event_dim = 2
 
     def check(self, value):
-        # Assumes that the matrix or batch of matrices in value are symmetric
-        # info == 0 means no error, that is, it's SPD
-        return torch.linalg.cholesky_ex(value).info.eq(0).unsqueeze(0)
+        return torch.full(
+            size=value.shape[:-2],
+            fill_value=(value.shape[-2] == value.shape[-1]),
+            dtype=torch.bool,
+            device=value.device
+        )
+
+
+class _Symmetric(_Square):
+    """
+    Constrain to Symmetric square matrices.
+    """
+
+    def check(self, value):
+        # square_check = super().check(value)
+        # if not square_check.all():
```

Comment:
Why are these lines commented out?

Reply:
It keep raises CI error and I wanted to find the cause.
This part
```
        return value.eq(value.mT).all(-2).all(-1)
```
is the problem and I am figuring out how to fix it.

Reply:
> > FYI, after merging this PR, I will try to fix the reverted PR, #68644 and merge it. Then, we can work on #68588 more neatly.
> 
> I would suggest moving the changes to this PR itself (since these are all related) so that we can add tests here.

I will move the changes in #68644 to this PR for now.